pennylane/ops/op_math/controlled_class.py

```diff
@@ -408,6 +410,16 @@ def __new__(cls, *_, **__):
         # overrides dispatch behavior of ``Controlled``
         return object.__new__(cls)
 
+    # pylint: disable=too-many-function-args
+    def __init__(
+        self, base, control_wires, control_values=None, work_wires=None, do_queue=True, id=None
+    ):
+        super().__init__(base, control_wires, control_values, work_wires, do_queue, id)
+        # check the grad_recipe validity
+        if self.grad_recipe is None:
+            # Make sure grad_recipe is an iterable of correct length instead of None
+            self.grad_recipe = [None] * self.num_params
```

Comment:
What is this for?

Reply:
I forgot to add this earlier.  Without it, we can't use parameter-shift even though the `parameter_frequencies` are defined.